scripts/quests/windurst/One_Good_Deed.lua

```diff
@@ -0,0 +1,145 @@
+                    if option == 1 then
+                        quest:begin(player)
+                    end
+                end,
+            },
+        }
+    },
+    {
+        check = function(player, status, vars)
+            return status == QUEST_ACCEPTED
+        end,
+
+        [xi.zone.PORT_WINDURST] =
+        {
+            ['Chipmy-Popmy'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 2 and player:hasKeyItem(xi.ki.DEED_TO_PURGONORGO_ISLE) then
+                        return quest:progressEvent(595, 0, xi.ki.DEED_TO_PURGONORGO_ISLE)
+                    elseif quest:getVar(player, 'Prog') == 4 then
+                        return quest:progressEvent(597) -- Complete Quest
+                    end
+                end,
+            },
+            onEventFinish =
+            {
+                [595] = function(player, csid, option, npc)
+                    quest:setVar(player, 'Prog', 3)
+                end,
+                [597] = function(player, csid, option, npc)
+                    quest:complete(player)
+                end,
+            },
+
+        },
+        [xi.zone.BIBIKI_BAY] =
+        {
+            ['qm_deed'] =
+            {
+                onTrigger = function(player, npc)
+                    if quest:getVar(player, 'Prog') == 0 then
+                        player:messageSpecial(bibikiID.text.YOU_ARE_NOT_ALONE)
+                        npcUtil.popFromQM(player, npc, bibikiID.mob.PERIFOOLS, {claim = true, hide = 0})
+                    elseif quest:getVar(player, 'Prog') == 1 then
+                        return quest:progressEvent(34)
+                    end
+                end,
+            },
+
+            ['Perifool'] =
+            {
+                onMobDeath = function(mob, player, isKiller, firstCall)
+                    if quest:getVar(player, 'Prog') == 0 then
+                        quest:setVar(player, 'Prog', 1)
+                    end
+                end,
+            },
+
+            onEventFinish =
+            {
+                -- Killing mobs set prog to 1
+                [34] = function(player, csid, option, npc)
+                    npcUtil.giveKeyItem(player, xi.ki.DEED_TO_PURGONORGO_ISLE)
+                    quest:setVar(player, 'Prog', 2)
+                end,
+            },
+        },
+        [xi.zone.BONEYARD_GULLY] =
+        {
+            onZoneIn =
+            {
+                function(player, prevZone)
+                    if quest:getVar(player, 'Prog') == 3 then
+                        return 8
+                    end
+                end,
+            },
+
+            onEventUpdate =
+            {
+                [8] = function(player, csid, option, npc)
+                    print(option)
+                    if option == 100 then
+                        quest:progressEvent(8, 11, xi.ki.DEED_TO_PURGONORGO_ISLE)
```

Comment:
do these 2 return the cs correctly? im not sure if this works or not and may need to be changed to player:startCutscene()

Reply:
Ah. That would make sense. The KI wasn't being referenced in the cutscene. Thank you